corehq/apps/app_manager/fields.py

```diff
@@ -396,8 +391,8 @@ def _get_applications_by_type(self, as_dict=True):
 
     @staticmethod
     def _map_chosen_by_choice_as_dict(chosen_by_choice):
-        for k, v in chosen_by_choice.items():
-            chosen_by_choice[k] = map(lambda f: f._asdict(), v)
+        for k, v in list(chosen_by_choice.items()):
```

Comment:
`chosen_by_choice` is being mutated inside the loop, so I thought it would break in Python 3 without `list(...)`. However, it does appear to work, probably because it's assigning to the same key:
```
Python 3.5.1 (default, Mar  2 2016, 03:38:02)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = {1: 2, 3: 4, 5: 6}
>>> for k, v in x.items():
...     x[k] = v ** 2
...
>>> x
{1: 4, 3: 16, 5: 36}
```
Indeed [the documentation](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects) implies that it's safe since we are not adding or deleting entries: "Iterating views while adding or deleting entries in the dictionary may raise a `RuntimeError` or fail to iterate over all entries."